app/models/concerns/padawan_run.rb

```diff
@@ -11,7 +11,7 @@ def improvements_towards(better_run)
       return improvements if collapsed_splits.size != better_run.collapsed_splits.size
 
       collapsed_splits.each.with_index do |split, i|
-        if better_run.splits[i].duration < split.duration && !split.reduced?
+        if better_run.splits[i].duration < split.duration && !split.reduced? && (split.duration - better_run.splits[i].duration) > 1
```

Comment:
+1 i like this change. However this can be simplified by changing the first check to `(better_run.splits[i].duration + 1) < split.duration` rather than having both checks